pkg/virt-controller/watch/migration.go

```diff
@@ -466,6 +470,22 @@ func (c *MigrationController) createTargetPod(migration *virtv1.VirtualMachineIn
 	// TODO libvirt requires unique host names for each target and source
 	templatePod.Spec.Hostname = ""
 
+	// If cpu model is "host model" allow migration only to nodes that supports this cpu model
+	if cpu := vmi.Spec.Domain.CPU; cpu != nil && cpu.Model == virtv1.CPUModeHostModel {
+		obj, exists, err := c.nodeInformer.GetStore().GetByKey(vmi.Status.NodeName)
+		node := obj.(*k8sv1.Node)
+		if err != nil {
+			return fmt.Errorf("cannot get nodes to migrate VMI with host-model CPU. error: %v", err)
+		} else if !exists {
+			return fmt.Errorf("node \"%s\" assosiated with vmi \"%s\" does not exist", vmi.Status.NodeName, vmi.Name)
+		}
+
+		err = prepareNodeSelectorForHostCpuModel(node, templatePod)
```

Comment:
I think we should fail the migration if we can't find the host model labels on the node. There is nothing really we can wait for if these labels don't exist on the node.

Reply:
> Great work! Great to see this is coming together.
> From my side, I think we are missing a feature gate. I think we need to block migrations with host-model and only allow migration if the feature gate is enabled.
> 
> I see that HCO[1] sets a non-existing feature-gate `WithHostModelCPU` for some reason...
> We can actually implement it.
> 
> I think that we also should fail the migration early ([somewhere here](https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-controller/watch/migration.go#L399)) in the case that no host-model labels were set on a source node.
> [1][https://github.com/kubevirt/hyperconverged-cluster-operator/blob/main/pkg/controller/operands/kubevirt.go#L108`](https://github.com/kubevirt/hyperconverged-cluster-operator/blob/main/pkg/controller/operands/kubevirt.go#L108%60)

@vladikr Sure, done!
A few questions though:
1. I've changed the name of the variable to `HostModelCPUFeatureGate` to align with our naming convention there. Do you think it's better to call it `WithHostModelCPU` to align with HCO? I tend to think it's more confusing than the first option.
2. Is there an easy way to get the name of this feature gate (`WithHostModelCPU`) from HCO? I'm afraid that if they change the name we won't be aligned.

Reply:
> I think we should fail the migration if we can't find the host model labels on the node. There is nothing really we can wait for if these labels don't exist on the node.

Got you.
Now the migration will fail if host-model labels don't exist for the node (and VMI is using host-model mode), before the migration is scheduled.

Reply:
> I don't think I understand why these functions are needed?
@iholder-redhat could you please explain?

Sorry, these are leftovers that are now deleted. Ignore them.
(used `nodeInformer` wrong, thanks to @xpivarc it's fixed now)